app/models/game_tag.rb

```diff
@@ -0,0 +1,8 @@
+class GameTag < ActiveRecord::Base
+  belongs_to :game
+  belongs_to :tag
+
+  validates :tag_id, presence: true
+  validates :game_id, presence: { message: "already has this tag" }
+  validates :tag_id, uniqueness: { scope: :game_id }
```

Comment:
We've mentioned this a few times. Please pay attention to prior comments. Attention to detail in pull request is a big deal in the real world.

You don't need `tag_id`, `game_id` here. You can validate directly against `:tag` and `:game`